keras/layers/normalization.py

```diff
@@ -57,16 +57,6 @@ def build(self):
         self.running_mean = K.zeros(input_shape)
         self.running_std = K.ones(input_shape)
 
-        # initialize self.updates: batch mean/std computation
-        X = self.get_input(train=True)
-        m = K.mean(X, axis=0)
-        std = K.mean(K.square(X - m) + self.epsilon, axis=0)
-        std = K.sqrt(std)
-        mean_update = self.momentum * self.running_mean + (1-self.momentum) * m
-        std_update = self.momentum * self.running_std + (1-self.momentum) * std
-        self.updates = [(self.running_mean, mean_update),
-                        (self.running_std, std_update)]
-
```

Comment:
Well, the input of BatchNormalization Layer can change after the `build` method was called. For example if the layer is included in a nested Sequential / Graph models as described in #1386. 
Having these lines in `build` results in a DisconnectedInputError since the `self.updates` rules use the wrong input.
In short there is no guarantee that `self.get_input` returns the same input in `build` and `get_output`.